src/engine/strat_engine/liminal/setup.rs

```diff
@@ -38,7 +38,19 @@ use crate::{
 /// is returned.
 pub fn get_bdas(infos: &HashMap<DevUuid, &LStratisInfo>) -> StratisResult<HashMap<DevUuid, BDA>> {
     fn read_bda(info: &LStratisInfo) -> StratisResult<BDA> {
-        BDA::load(&mut OpenOptions::new().read(true).open(&info.ids.devnode)?)?.ok_or_else(|| {
+        let f = &mut OpenOptions::new().read(true).open(&info.ids.devnode)?;
```

Comment:
The ```collect``` method is doing something pretty sophisticated in the code below: if read_bda() returns an error on reading any BDA, then that is the error that will be returned from ```get_bdas()```. So, whatever error you do return from ```read_bda()``` it should be at least as identifying as the BDA one, which says what the expected ids of the device are.

Your choices of what to interpret as an error an what not to are exactly correct.